src/motor_protection.h

```diff
@@ -0,0 +1,24 @@
+#ifndef MOTOR_PROTECTION_H
+#define MOTOR_PROTECTION_H
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+typedef struct {
+    float t;
+    float t_max;
+    float t_ambient;
+    float r_th;
```

Comment:
What are those coefficient ? I am guessing thermal capacity and thermal resistance, but maybe a comment with the unit too would be good.